### PR TITLE
feat(browsersync): added browswersync to watch command

### PR DESCRIPTION
**Pull request by dperrera**
I added browser-sync as we were discussing in #48. BrowserSync proxies Reptar's server and watches the source directory, which feels pretty good to me.

This is my first real commit to an open source project and I'm sure I messed something up. Don't hesitate to let me know how this could be better.

### CODE REVIEW

**Comment by hswolff** [comment]
This is super exciting! I'll update when I have a chance to check out and test it! Thank you!

File: lib/cli/watch.js [hunk 1]
```diff
@@ -17,9 +18,14 @@ export default async function watch(options = {}) {
...
   await server.start();
-
   activity.end(startActivity);
 
   process.stdout.write('\n');
   log.info('Server running at:', server.server.info.uri);
+
+  const browserSync = bsCreate();
+  browserSync.init({
+    files: `${reptar.config._raw.path.source}/**/*`,
```
Comment (hswolff) [inline lib/cli/watch.js]: We should use the `config.get()` method to access the value.

Comment (dperrera) [inline lib/cli/watch.js]: No problem! I was also thinking about a CLI flag. I pretty much always want to use browser-sync  but I could imagine a scenario where someone might only want to watch. What should the flag be? `-bs`, `-l`?

Comment (hswolff) [inline lib/cli/watch.js]: `--browser-sync` with `-bs` as a shortcut?

Comment (dperrera) [inline lib/cli/watch.js]: I like it. I'll implement that shortly!

**Comment by dperrera** [comment]
Of course, now that I updated the PR I have another idea. Browser Sync has a bunch of options that it will accept and it'd be pretty nice to pass them in from `reptar.config.js`. Does that sound interesting?

**Comment by hswolff** [comment]
Hmm...what options do you see needing configuration? I'm not against it but hesitant to add it just because we can.

File: lib/cli/watch.js [hunk 2]
```diff
@@ -17,9 +18,16 @@ export default async function watch(options = {}) {
 
   const server = new Server(reptar);
   await server.start();
-
   activity.end(startActivity);
 
   process.stdout.write('\n');
   log.info('Server running at:', server.server.info.uri);
+
+  if (reptar.options.browserSync) {
```
Comment (hswolff) [inline lib/cli/watch.js]: Did you test disabling browserSync? I think you'd want the options object given to the watch function in here.

Comment (dperrera) [inline lib/cli/watch.js]: I changed this to get the options from the watch function. That makes perfect sense. 

I'm not sure I follow about disabling browserSync. I uninstalled it globally on my machine, which did cause a problem. But I just needed to `npm install` in my test Reptar project and it worked like a charm.

**Comment by dperrera** [comment]
Browser Sync has a whole bunch of [options](https://browsersync.io/docs/options). The most common  uses would be changing the port and configuring the browser it opens but maybe we should wait for people request that feature? We'd only need to add one line of code to `config-schema.js` and one line to `watch.js` to make it happen but I understand wanting to avoid needless bloat.

**Comment by hswolff** [comment]
Yeah let's add it later if we need it. This LGTM! Can you rebase and squash the commits down to one please?

**Comment by dperrera** [comment]
Squashed and ready to go!

**Comment by hswolff** [comment]
Woohoo! Thank you!

**Comment by jmsmrgn** [comment]
Can we please get this sweet feature published? Keep making cool shit!

**Comment by hswolff** [comment]
Just published in version `v3.3.0`! Thanks for the support!

**Pull request merged**